dowhy/gcm/distribution_change_robust.py

```diff
@@ -0,0 +1,915 @@
+            X_calib, T_calib = None, None
+
+            if calibrator is not None:
+                if calib_size == 0.0:
+                    raise ValueError(
+                        "For calibration, calib_size should be either positive and smaller than the number of samples or a float in the (0, 1) range."
+                    )
+
+                X_calib, X_train, _, y_train, T_calib, T_train = train_test_split(
+                    X_train, y_train, T_train, train_size=calib_size, stratify=T_train, random_state=split_random_state
+                )
+
+            w_eval, w_train, w_calib = None, None, None
+
+        else:
+            w = np.concatenate((old_data[sample_weight].values.flatten(), new_data[sample_weight].values.flatten()))
+
+            X_train, X_eval, y_train, y_eval, T_train, T_eval, w_train, w_eval = train_test_split(
+                X, y, T, w, train_size=train_size, stratify=T, random_state=split_random_state
+            )
+            X_calib, T_calib, w_calib = None, None, None
+
+            if calibrator is not None:
+                if calib_size == 0.0:
+                    raise ValueError(
+                        "For calibration, calib_size should be either positive and smaller than the number of samples or a float in the (0, 1) range."
+                    )
+
+                X_calib, X_train, _, y_train, T_calib, T_train, w_calib, w_train = train_test_split(
+                    X_train,
+                    y_train,
+                    T_train,
+                    w_train,
+                    train_size=calib_size,
+                    stratify=T_train,
+                    random_state=split_random_state,
+                )
+
+        def set_func(C):
+            return ThetaC(C, warn_th=0.0).est_theta(
+                X_eval,
+                y_eval,
+                T_eval,
+                X_train,
+                y_train,
+                T_train,
+                w_eval=w_eval,
+                w_train=w_train,
+                method=method,  # One of 'regression', 're-weighting', 'MR',
+                regressor=regressor,
+                regressor_args=regressor_args,
+                regressor_kwargs=regressor_kwargs,
+                regressor_fit_kwargs=regressor_fit_kwargs,
+                classifier=classifier,
+                classifier_args=classifier_args,
+                classifier_kwargs=classifier_kwargs,
+                classifier_fit_kwargs=classifier_fit_kwargs,
+                calibrator=calibrator,
+                X_calib=X_calib,
+                T_calib=T_calib,
+                w_calib=w_calib,
+                calibrator_args=calibrator_args,
+                calibrator_kwargs=calibrator_kwargs,
+                calibrator_fit_kwargs=calibrator_fit_kwargs,
+                all_indep=all_indep,
+                crop=crop,
+            )[0]
+
+        attributions = estimate_shapley_values(set_func, len(sorted_var_names), shapley_config)
+
+    if xfit:
```

Comment:
Put this part in the else branch. Also, rather have the order `if xfit ... else: ...`. It is easier to follow if the `true` case happens first.

Reply:
Fixed.